lib/core/connector_job.rb

```diff
@@ -34,12 +36,33 @@ def self.pending_jobs(connectors_ids: [], page_size: DEFAULT_PAGE_SIZE)
       fetch_jobs_by_query(query, page_size)
     end
 
-    def self.orphaned_jobs(_page_size = DEFAULT_PAGE_SIZE)
-      []
+    def self.orphaned_jobs(page_size = DEFAULT_PAGE_SIZE)
+      connector_ids = ConnectorSettings.fetch_all_connectors.map(&:id)
+      query = { bool: { must_not: { terms: { 'connector.id': connector_ids } } } }
+      fetch_jobs_by_query(query, page_size)
     end
 
-    def self.stuck_jobs(_page_size = DEFAULT_PAGE_SIZE)
-      []
+    def self.delete_jobs(jobs)
+      query = { terms: { '_id': jobs.map(&:id) } }
+      ElasticConnectorActions.delete_jobs_by_query(query)
```

Comment:
I don't think we should ever delete anything. Mark as failed with cause "orphaned"?

Reply:
Oh, right, these are jobs that are there while connector record has been deleted, disregard this comment.